pkg/cortex/modules.go

```diff
@@ -536,6 +547,53 @@ func (t *Cortex) stopMemberlistKV() (err error) {
 	return nil
 }
 
+func (t *Cortex) initDataPurger(cfg *Config) (err error) {
+	if !cfg.DataPurgerConfig.EnablePurger {
+		return nil
+	}
+
+	var indexClient chunk.IndexClient
+	indexClient, err = storage.NewIndexClient(cfg.Storage.DeleteStoreConfig.Store, cfg.Storage, cfg.Schema)
+	if err != nil {
+		return
+	}
+
+	var deleteStore *chunk.DeleteStore
+	deleteStore, err = chunk.NewDeleteStore(cfg.Storage.DeleteStoreConfig, indexClient)
+	if err != nil {
+		return
+	}
+
+	var storageClient chunk.StorageClient
+	storageClient, err = storage.NewStorageClient(cfg.DataPurgerConfig.ObjectStoreType, cfg.Storage)
+	if err != nil {
+		return
+	}
+
+	t.dataPurger, err = purger.NewDataPurger(cfg.DataPurgerConfig, deleteStore, t.store, storageClient)
+	if err != nil {
+		return
+	}
+
+	go t.dataPurger.Run()
+
+	var deleteRequestHandler *purger.DeleteRequestHandler
+	deleteRequestHandler, err = purger.NewDeleteRequestHandler(deleteStore)
+	if err != nil {
+		return
+	}
+
+	t.server.HTTP.Path("/delete_series").Handler(t.httpAuthMiddleware.Wrap(http.HandlerFunc(deleteRequestHandler.AddDeleteRequestHandler)))
+	t.server.HTTP.Path("/get_all_delete_requests").Handler(t.httpAuthMiddleware.Wrap(http.HandlerFunc(deleteRequestHandler.GetAllDeleteRequestsHandler)))
```

Comment:
`/get_all_delete_requests` is a somewhat clunky url path. Instead of using a separate path can you just reuse the `/api/v1/admin/tsdb/delete_series` path and just call this handler function when a `GET` method is used. According to the [Prometheus Docs](https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series) the `/delete_series` should only accept `POST` or `PUT` requests anyway.